### ISSUE TITLE
Global plugin v5.4.18 vs v5.4.19 

### ISSUE THREAD

**Issue by venkateshwarlu-bavandla** [Automattic/mongoose#7916]
Hi the following code behaves different in both versions v5.4.18 and v5.4.19
```javascript
var mongoose = require('mongoose')

 mongoose.plugin(function updateDateFields (schema, options) {
    if (!schema.path('atr')) {
      schema.add({ atr: Date })
    }
  
    schema.pre('validate', function (next) {
      console.log('validate hook called !')
      this.atr = Date.now()
      next()
    })
  })


let Schema = mongoose.Schema

var eventSchema = new Schema({
    url:  { type: String }
})


eventSchema.add({headers: {
    type: [{
      name: { type: String},
      value: { type: String},
      _id: false
    }],
    default: undefined
  }})
var Event = mongoose.model('Event', eventSchema)

var eventObj = new Event({
    url: 'google.com',
    headers: [{name: "q1", value:'1'},{name: 'Q2', value:'2'}]

});

eventObj.validate(function() {
    console.log(eventObj)
})
```
Output with  v5.4.18 :
```javascript
validate hook called !
validate hook called !
validate hook called !
{ _id: '5d0c741c84e784213c3dde0f',
  url: 'google.com',
  headers: [ { name: 'q1', value: '1' }, { name: 'Q2', value: '2' } ],
  atr: '2019-06-21T06:07:24.465Z' }
```
Output with  v5.4.19:
```javascript
validate hook called !
validate hook called !
validate hook called !
{ _id: '5d0c62fa47df581e39b95490',
  url: 'google.com',
  headers:
   [ { name: 'q1', value: '1', atr: '2019-06-21T04:54:18.192Z' },
     { name: 'Q2', value: '2', atr: '2019-06-21T04:54:18.192Z' } ],
  atr: '2019-06-21T04:54:18.189Z' }
```

In the v5.4.19 version "atr" field includes in the each of sub docs also, But in my case only in the root level we need "atr" field. 

Can any body suggest how to achieve the old behaviour with newer version. I found that in v5.4.19 some changes done on global plug in 
https://github.com/Automattic/mongoose/blob/master/History.md#5419--2019-03-11

PS: Is this bug?

**Comment by venkateshwarlu-bavandla** [comment]
This issue exist on latest version v5.6.0

**Comment by vkarpov15** [comment]
This is by design - global plugins apply to _all_ schemas by default, including nested schemas. There was a bug in 5.4.x before 5.4.19 that prevented some changes to nested schemas in global plugins from getting applied, which we fixed in #7572 . We'll add an option for v5.6.2: `mongoose.set('applyPluginsToChildSchemas', false)`
